keel-clouddriver/src/main/kotlin/com/netflix/spinnaker/keel/clouddriver/ImageService.kt

```diff
@@ -259,6 +213,35 @@ class ImageService(
   }
 }
 
+
+/**
+ * Find the latest properly tagged images in [account] for each of [regions] using
+ * [ImageService.getLatestNamedImageForAppVersionInRegion] in parallel for each region.
+ *
+ * In many cases all the values in the resulting map will be the same [NamedImage] instance, but
+ * this may not be the case if images were baked separately in each region.
+ *
+ * The resulting map will contain no entry for regions where an image is not found. The calling
+ * code must check this if it requires all regions to be present.
+ */
+suspend fun ImageService.getLatestNamedImageForAppVersionInRegions(
+  appVersion: AppVersion,
+  account: String,
+  regions: Collection<String>
+): Map<String, NamedImage> = coroutineScope {
+  regions.associateWith { region ->
+    async {
+      getLatestNamedImageForAppVersionInRegion(
+        appVersion = appVersion,
+        account = account,
+        region = region
+      )
+    }
+  }
+    .mapValues { (_, it) -> it.await() }
+    .filterNotNullValues()
```

Comment:
Method to find images in multiple regions. It's just doing a nice little coroutine-ized async call to `getLatestNamedImageForAppVersionInRegion` for each region. Most often the result will be the same for all regions.

The advantage of doing it this way is that the calling code doesn't have to consider what regions are supported by what image, it can just key off the map and not care if they're all the same image, all different ones, or somewhere in between.

I decided to always return whatever regions we _do_ find, rather than returning nothing if we don't find them all as it's less confusing. This just means the calling code needs to check whether all regions are in the result map (if it cares).

I implemented this as an extension function because that way the tests can use stubs for the individual region calls. In some cases that simpler, but it also has the advantage that it's actually exercising this parallelization logic.